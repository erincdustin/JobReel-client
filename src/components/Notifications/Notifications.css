.notifications-box{
  background: #3C3230;
  color: white;
  padding: 10px;
  border-radius: 10px;
  grid-column-start: 4;
  grid-column-end: 9;
  height: 250px;
}

.notifications h2{
 text-align: center;
}