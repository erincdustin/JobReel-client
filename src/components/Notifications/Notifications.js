import React from 'react';
import './Notifications.css';
import JobNotification from '../JobNotification/JobNotification';
import ContactNotification from '../ContactNotification/ContactNotification';
import jobReelContext from '../../context/JobReelContext';

class Notifications extends React.Component {

  static contextType = jobReelContext

  renderDatesApplied(){
    let jobs = Array.from(this.context.savedJobs)
    if(this.context.savedJobs === []){
      return ''
    } else {
      let applied = jobs.map((job, i) => {
        return <JobNotification 
        key={i} 
        user_id={job.user_id} 
        job_id={job.job_id} 
        city={job.city} 
        state={job.state} 
        url={job.url} 
        status={job.status} 
        desc={job.description} 
        date_added={job.date_added} 
        company={job.company} 
        title={job.job_title} 
        date_applied={job.date_applied} 
        notification={job.notification}/>
      })
      return <div>{applied}</div>
    }
  }

  renderContactsConnected(){
    let contacts = Array.from(this.context.contacts)
    if(this.context.contacts === []){
      return ''
    } else {
      let contacted = contacts.map((contact, i) => {
        return <ContactNotification 
        key={i} 
        contact_id={contact.contact_id} 
        name={contact.contact_name} 
        job_title={contact.job_title}
        company={contact.company}
        linkedin={contact.linkedin}
        comments={contact.comments}
        email={contact.email} 
        connected={contact.connected} 
        date_added={contact.date_added}
        date_connected={contact.date_connected} 
        user_id={contact.user_id} 
        notification={contact.notification} />
      })
      return <div>{contacted}</div>
    }
  }

  render() {
    return (
      <div className="notifications-box">
        <div className='notification-header'>
          <h3>Notifications</h3>
        </div>
        <div className="notifications">
            {this.renderDatesApplied()}
            {this.renderContactsConnected()}
        </div>
      </div>
    )
  }
}

export default Notifications;