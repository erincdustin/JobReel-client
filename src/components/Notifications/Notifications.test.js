import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Notifications from './Notifications'
import PropTypes from 'prop-types'

class ContextProvider extends React.Component{
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      contacts: [{
        contact_id: 1,
        contact_name: 'Contact 1',
        job_title: 'Engineer',
        company: 'Company1',
        email: 'email@email.com',
        linkedin: 'http://www.linkedin.com/person1',
        comments: 'Contact 1 comments',
        date_added: '2019-07-03T19:26:38.918Z',
        connected: false,
        user_id: 1,
        date_connected: '2019-08-03T19:26:38.918Z',
        notification: true
      }],
      savedJobs: [{
        job_id: 2,
        user_id: 1,
        job_title: 'UI Designer',
        company: 'Company B',
        city: 'Austin',
        state: 'TX',
        date_added: '2019-08-14T17:18:19.306Z',
        url: 'http://www.google.com',
        description: 'UI job',
        status: 'Interested',
        date_applied: '2019-08-16T17:18:19.306Z',
        notification: true
      }]
    }
  })

  render(){
    return this.props.children
  }
}

describe('Notifications smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Notifications />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('Notifications snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><Notifications /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})