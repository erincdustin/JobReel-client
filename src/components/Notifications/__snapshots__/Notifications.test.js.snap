// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications snapshot test renders the UI as expected 1`] = `<Notifications />`;
